### PR TITLE
Introduce Binary Distribution Workflow

### PR DESCRIPTION
**Pull request by aleneum**
This commit introduces a workflow to build and distribute Ubuntu packages for `trusty` and `xenial`. The maintainer/user may use `dh` (source and binary) or `CPack` to build packages.
I also documented the workflow to push packages to the launchpad [PPA](https://launchpad.net/~iclcv/+archive/ubuntu/icl). Whoever has a Ubuntu ONE/Launchpad account can give me a shout to be added to the iclcv team.

### Changes

* split cmake targets into several components
  - this eases bundling of `CPack` groups for Debian packages (and binary installers if we want to go that route)
* add `packaging`; includes ...
  - ... a script to configure the ICL for packaging
  - ... Docker recipe for a ICL packaging environment (trusty)
  - ... debian rules and control file templates; will be deployed to the source root when `BUILD_DEBS` is enabled
  - ... packaging.md for a brief description of the packaging process

### CODE REVIEW

**Comment by aleneum** [comment]
as discussed in #1

File: README.md [hunk 1]
```diff
@@ -169,18 +170,48 @@ We **strongly** recommend to have at least Qt support when building ICL.
...
+
+Ubuntu packages can be obtained from the [Releases](https://github.com/iclcv/icl/releases) page or by using the [ICL PPA](https://launchpad.net/~iclcv/+archive/ubuntu/icl):
+
+```bash
+sudo add-apt-repository ppa:iclcv/icl
+sudo apt-get update
+sudo apt-get install icl-tools
+```
+
+The ICL built is divided into four packages:
```
Comment (rhaschke) [inline README.md]: built -> build

File: README.md [hunk 2]
```diff
@@ -169,18 +170,48 @@ We **strongly** recommend to have at least Qt support when building ICL.
...
 $ brew install icl  
 # base features plus additional functionality based on rsb, protobuf, freenect, pcl and bullet
 $ brew install icl --with-extra
 # extra features plus libusb, zmq and openni support  
 brew install icl --with-full
 ```
+
+## Questions, feedback, issues or feature requests?
+
+For bug reports and other issues, please open an issue on GitHub.
```
Comment (rhaschke) [inline README.md]: Add the link to GitHub here

File: packaging/scripts/ubuntu-build.sh [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+      -DBUILD_WITH_LIBFREENECT=ON \
+      -DBUILD_WITH_V4L=ON \
+      -DBUILD_WITH_LIBUSB=ON -DBUILD_WITH_ZMQ=ON \
+      -DBUILD_WITH_OPENCL=ON \
+      -DBUILD_WITH_OPENNI=ON -DBUILD_WITH_PCL=ON \
+      -DBUILD_EXAMPLES=ON -DBUILD_DEMOS=ON \
+      -DBUILD_APPS=ON -DBUILD_DEBS=ON ..
+
+
+# -DBUILD_WITH_IPP=ON -DBUILD_WITH_MKL=ON (Custom Repos)
```
Comment (rhaschke) [inline packaging/scripts/ubuntu-build.sh]: remove commented stuff?

**Review by rhaschke — APPROVED** [review]
Really cool!

**Comment by aleneum** [comment]
I agree to all remarks. Thanks!

**Pull request merged**